Package Request: projectM visualizer

Edit: ignore this template, new one posted below.

This is so far what I've got for a template for projectM, I've borrowed some of the pkg_install() bits from archlinux, what's the void-equivalent for DESTDIR="$pkgdir" in this case? 

edit: I've just realized projectm is no longer using cmake & qt4 in their new version, ignore that.

```
# Template file for 'projectm'
pkgname=projectm
version=3.1.0
revision=1
build_style=gnu-configure
hostmakedepends="pkg-config qt-devel-tools autoconf automake libtool glm"
makedepends="pulseaudio-devel alsa-lib-devel glew-devel ftgl-devel libXext-devel SDL2-devel
 gtkglext-devel jack-devel qt-devel libvisual-devel MesaLib-devel glu-devel"
depends="hicolor-icon-theme desktop-file-utils dejavu-fonts-ttf libGL"
short_desc="Cross-platform music visualization milkdrop-compatible"
maintainer="Johnny-Mac <johnny-mac@linuxmail.org>"
license="LGPL-2.1-or-later"
homepage="https://projectm.sourceforge.net"
distfiles="https://github.com/${pkgname}-visualizer/${pkgname}/archive/v${version}.tar.gz"
checksum=8ddf99dc141e554d76a2ecf8d8c1fe7996051e30fa8dd4fc133ce024c8defba7
pre_configure() {
	./autogen.sh
}
projectm_package() {
	short_desc=" - projectm Music visualizer which uses 3D accelerated iterative image based rendering"
	pkg_install() {
		DESTDIR="$pkgdir" make -C "build/src/NativePresets" install
		DESTDIR="$pkgdir" make -C "build/src/libprojectM" install
		DESTDIR="$pkgdir" install -Dm644 "${srcdir}/projectM-complete-$pkgver-Source/COPYING.txt" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
	}
}
projectm-libvisual_package() {
	short_desc=" - projectm plugin for XMMS"
	depends="projectm libvisual libgcc"
	pkg_install() {
		DESTDIR="$pkgdir" make -C "build/src/projectM-libvisual" install
		DESTDIR="$pkgdir" install -Dm644 "${srcdir}/projectM-complete-$pkgver-Source/COPYING.txt" "$pkgdir/usr/share/licenses/${pkgname}/LICENSE"
	}
}
projectm-pulseaudio_package() {
	short_desc=" - projectm support for PulseAudio"
	depends="projectm-qt pulseaudio"
	pkg_install() {
		DESTDIR="$pkgdir" make -C "$srcdir/build/src/projectM-pulseaudio" install
		DESTDIR="$pkgdir" install -Dm644 "${srcdir}/projectM-complete-$pkgver-Source/COPYING.txt" "$pkgdir/usr/share/licenses/${pkgname}/LICENSE"
	}
}
projectm-qt_package() {
	short_desc=" - Qt bindings for projectm'
	depends=(projectm qt libGL"
	pkg_install() {
		DESTDIR="$pkgdir" make -C "$srcdir/build/src/projectM-qt" install
		DESTDIR="$pkgdir" install -Dm644 "${srcdir}/projectM-complete-$pkgver-Source/COPYING.txt" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
	}
}
```

```
# Template file for 'projectm'
pkgname=projectm
version=3.1.0
revision=1
build_style=gnu-configure
configure_args="--enable-qt"
hostmakedepends="pkg-config qt5-host-tools autoconf automake libtool glm"
makedepends="pulseaudio-devel alsa-lib-devel glew-devel ftgl-devel libXext-devel SDL2-devel
 gtkglext-devel jack-devel qt5-tools-devel libvisual-devel MesaLib-devel glu-devel"
depends="hicolor-icon-theme desktop-file-utils dejavu-fonts-ttf libGL"
short_desc="Cross-platform music visualization milkdrop-compatible"
maintainer="Johnny-Mac <johnny-mac@linuxmail.org>"
license="LGPL-2.1-or-later"
homepage="https://projectm.sourceforge.net"
distfiles="https://github.com/${pkgname}-visualizer/${pkgname}/archive/v${version}.tar.gz"
checksum=8ddf99dc141e554d76a2ecf8d8c1fe7996051e30fa8dd4fc133ce024c8defba7
pre_configure() {
        ./autogen.sh
}
```

This is so far what I've got, but it fails to build unfortunately...

```
Shader.cpp  -fPIC -DPIC -o .libs/libRenderer_la-Shader.o
In file included from Renderable.cpp:3:
Renderable.hpp: In constructor 'TypeIdPair::TypeIdPair(const string&, const string&)':
Renderable.hpp:162:62: warning: declaration of 'id2' shadows a member of 'TypeIdPair' [-Wshadow]
  TypeIdPair(const std::string & id1, const std::string & id2): id1(id1), id2(id2) {}
                                                              ^
Renderable.hpp:164:14: note: shadowed declaration is here
  std::string id2;
              ^~~
Renderable.hpp:162:62: warning: declaration of 'id1' shadows a member of 'TypeIdPair' [-Wshadow]
  TypeIdPair(const std::string & id1, const std::string & id2): id1(id1), id2(id2) {}
                                                              ^
Renderable.hpp:163:14: note: shadowed declaration is here
  std::string id1;
              ^~~
Renderable.cpp: In constructor 'Shape::Shape()':
Renderable.cpp:86:15: warning: declaration of 'imageUrl' shadows a member of 'Shape' [-Wshadow]
   std::string imageUrl = "";
               ^~~~~~~~
In file included from Renderable.cpp:3:
Renderable.hpp:62:17: note: shadowed declaration is here
     std::string imageUrl;
                 ^~~~~~~~
Renderable.cpp: In member function 'virtual void MotionVectors::Draw(RenderContext&)':
Renderable.cpp:332:19: warning: unused variable 'lz' [-Wunused-variable]
     float lx, ly, lz;
                   ^~
mv -f .deps/libRenderer_la-Filters.Tpo .deps/libRenderer_la-Filters.Plo
/bin/sh ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -Wall -Wchar-subscripts -Wformat-security -Wmissing-declarations -Wpointer-arith -Wshadow -Wsign-compare -Wtype-limits -DDATADIR_PATH=\""/usr/share/projectM"\" -I../../../vendor  -include ../../../config.h -I../../../src/libprojectM/Renderer/hlslparser/src -I../../../src/libprojectM   -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe    -pthread -std=c++11 -MT libRenderer_la-TextureManager.lo -MD -MP -MF .deps/libRenderer_la-TextureManager.Tpo -c -o libRenderer_la-TextureManager.lo `test -f 'TextureManager.cpp' || echo './'`TextureManager.cpp
make[4]: *** [Makefile:679: libRenderer_la-ShaderEngine.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
mv -f .deps/libRenderer_la-Pipeline.Tpo .deps/libRenderer_la-Pipeline.Plo
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -Wall -Wchar-subscripts -Wformat-security -Wmissing-declarations -Wpointer-arith -Wshadow -Wsign-compare -Wtype-limits -DDATADIR_PATH=\"/usr/share/projectM\" -I../../../vendor -include ../../../config.h -I../../../src/libprojectM/Renderer/hlslparser/src -I../../../src/libprojectM -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe -pthread -std=c++11 -MT libRenderer_la-Renderable.lo -MD -MP -MF .deps/libRenderer_la-Renderable.Tpo -c Renderable.cpp -o libRenderer_la-Renderable.o >/dev/null 2>&1
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -Wall -Wchar-subscripts -Wformat-security -Wmissing-declarations -Wpointer-arith -Wshadow -Wsign-compare -Wtype-limits -DDATADIR_PATH=\"/usr/share/projectM\" -I../../../vendor -include ../../../config.h -I../../../src/libprojectM/Renderer/hlslparser/src -I../../../src/libprojectM -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe -pthread -std=c++11 -MT libRenderer_la-TextureManager.lo -MD -MP -MF .deps/libRenderer_la-TextureManager.Tpo -c TextureManager.cpp  -fPIC -DPIC -o .libs/libRenderer_la-TextureManager.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -Wall -Wchar-subscripts -Wformat-security -Wmissing-declarations -Wpointer-arith -Wshadow -Wsign-compare -Wtype-limits -DDATADIR_PATH=\"/usr/share/projectM\" -I../../../vendor -include ../../../config.h -I../../../src/libprojectM/Renderer/hlslparser/src -I../../../src/libprojectM -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe -pthread -std=c++11 -MT libRenderer_la-Shader.lo -MD -MP -MF .deps/libRenderer_la-Shader.Tpo -c Shader.cpp -o libRenderer_la-Shader.o >/dev/null 2>&1
mv -f .deps/libRenderer_la-Waveform.Tpo .deps/libRenderer_la-Waveform.Plo
mv -f .deps/libRenderer_la-Shader.Tpo .deps/libRenderer_la-Shader.Plo
mv -f .deps/libRenderer_la-Renderer.Tpo .deps/libRenderer_la-Renderer.Plo
mv -f .deps/libRenderer_la-MilkdropWaveform.Tpo .deps/libRenderer_la-MilkdropWaveform.Plo
mv -f .deps/libRenderer_la-Renderable.Tpo .deps/libRenderer_la-Renderable.Plo
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -Wall -Wchar-subscripts -Wformat-security -Wmissing-declarations -Wpointer-arith -Wshadow -Wsign-compare -Wtype-limits -DDATADIR_PATH=\"/usr/share/projectM\" -I../../../vendor -include ../../../config.h -I../../../src/libprojectM/Renderer/hlslparser/src -I../../../src/libprojectM -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe -pthread -std=c++11 -MT libRenderer_la-TextureManager.lo -MD -MP -MF .deps/libRenderer_la-TextureManager.Tpo -c TextureManager.cpp -o libRenderer_la-TextureManager.o >/dev/null 2>&1
mv -f .deps/libRenderer_la-TextureManager.Tpo .deps/libRenderer_la-TextureManager.Plo
make[4]: Leaving directory '/builddir/projectm-3.1.0/src/libprojectM/Renderer'
make[3]: *** [Makefile:640: all-recursive] Error 1
make[3]: Leaving directory '/builddir/projectm-3.1.0/src/libprojectM'
make[2]: *** [Makefile:443: all-recursive] Error 1
make[2]: Leaving directory '/builddir/projectm-3.1.0/src'
make[1]: *** [Makefile:513: all-recursive] Error 1
make[1]: Leaving directory '/builddir/projectm-3.1.0'
make: *** [Makefile:403: all] Error 2

```

--- a/srcpkgs/projectM/template
+++ b/srcpkgs/projectM/template
@@ -1,0 +1,16 @@
+# Template file for 'projectM'
+pkgname=projectM
+version=3.1.3
+revision=1
+build_style=gnu-configure
+configure_args="--datadir=/usr/libexec"
+hostmakedepends="pkg-config qt5-host-tools automake libtool which"
+makedepends="pulseaudio-devel alsa-lib-devel glew-devel ftgl-devel libXext-devel
+ SDL2-devel gtkglext-devel jack-devel qt5-tools-devel libvisual-devel glm
+ MesaLib-devel glu-devel"
+short_desc="Music visualization"
+maintainer="Andrew Benson <abenson+void@gmail.com>"
+license="LGPL-2.1-or-later"
+homepage="https://github.com/projectM-visualizer/projectm"
+distfiles="${homepage}/releases/download/v${version}/projectM-${version}.tar.gz"
+checksum=6e5879fad747f05bab348a6b2af8fe2f73ad6a901f0dfaa544e9e1b6cefbadc0

